nifi-nar-bundles/nifi-standard-services/nifi-hbase_1_1_2-client-service-bundle/nifi-hbase_1_1_2-client-service/src/main/java/org/apache/nifi/hbase/HBase_1_1_2_LookupService.java

```diff
@@ -0,0 +1,168 @@
+        super.init(config);
+        this.lookupProperties = new ArrayList<>();
+        this.lookupProperties.addAll(properties);
+        this.lookupProperties.add(TABLE_NAME);
+        this.lookupProperties.add(RETURN_CFS);
+        this.lookupProperties.add(RETURN_QFS);
+    }
+
+    @Override
+    public Optional lookup(Map coordinates) throws LookupFailureException {
+        byte[] rowKey = ((String)coordinates.get("rowKey")).getBytes();
+        try {
+            Map<String, Object> values = new HashMap<>();
+            try (Table table = this.connection.getTable(TableName.valueOf(tableName))) {
+                Get get = new Get(rowKey);
+                Result result = table.get(get);
+
+                for (byte[] fam : families) {
+                    NavigableMap<byte[], byte[]>  map = result.getFamilyMap(fam);
+                    for (Map.Entry<byte[], byte[]> entry : map.entrySet()) {
+                        if (qualifiers.contains(entry.getKey()) || qualifiers.size() == 0) {
+                            values.put(new String(entry.getKey()), new String(entry.getValue()));
+                        }
+                    }
+                }
+            }
+
+            if (values.size() == 1) {
+                return Optional.ofNullable(values.values().iterator().next());
+            } else if (values.size() > 1) {
+                final List<RecordField> fields = new ArrayList<>();
+                fields.add(new RecordField("key1", RecordFieldType.STRING.getDataType()));
+                fields.add(new RecordField("key2", RecordFieldType.STRING.getDataType()));
+                fields.add(new RecordField("key3", RecordFieldType.STRING.getDataType()));
+                final RecordSchema schema = new SimpleRecordSchema(fields);
+                return Optional.ofNullable(new MapRecord(schema, values));
+            } else {
+                throw new LookupFailureException(String.format("Nothing was found that matched the criteria for row key %s", coordinates.get("rowKey")));
+            }
+        } catch (IOException e) {
+            getLogger().error("Error occurred loading {}", new Object[] { coordinates.get("rowKey") }, e);
+            throw new LookupFailureException(e);
+        }
+    }
+
+    @Override
+    public Class<?> getValueType() {
+        return null;
+    }
+
+    @Override
+    public Set<String> getRequiredKeys() {
+        return REQUIRED_KEYS;
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    @OnEnabled
+    public void onEnabled(final ConfigurationContext context) throws InitializationException, IOException, InterruptedException {
+        super.onEnabled(context);
+
+        this.tableName = context.getProperty(TABLE_NAME).getValue();
+        String families = context.getProperty(RETURN_CFS).getValue();
+        String[] familiesSplit = families.split(",[\\s]*");
```

Comment:
It may be better in this case to use split(",") and then trim each result as you iterate over them, simply because as-is, it will strip out any leading white space but not trailing white space (i.e., if i used a value like "value1 , value2" then it would result in "value1 " (with a space) and "value2")